client/client.go

```diff
@@ -417,6 +418,13 @@ func (c *Client) getLocalMeta() error {
 			c.loadTargets(targets.Targets)
 		}
 	}
+
+	for fileName := range meta {
+		if roles.IsDelegatedTargetsManifest(fileName) {
+			c.localMeta[fileName] = meta[fileName]
```

Comment:
Besides, this PR just aims to rollback https://github.com/theupdateframework/go-tuf/commit/ed6788e710fc3093a7ecc2d078bf734c0f200d8d, where it wasn't verified, so I'm not sure we should do it in this PR

Reply:
Agreed with the majority consensus here

Reply:
Great, let us know how we'll proceed so @trishankatdatadog can move forward with backporting it to v0.3.0

Reply:
Would it be possible to deal with this in a follow up PR ? We took some time to investigate adding verification but the best options we found are not trivial require significant changes. IMO we should
1. Fix this bug (it's blocking us from updating beyond v0.3.0 and we're currently maintaining a fork https://github.com/DataDog/go-tuf)
2. Harden the verification

Reply:
Sounds reasonable to me. Let me look at whether this PR will work for everyone else right now.

Reply:
How is this a regression BTW? I don't see it in at least https://github.com/theupdateframework/go-tuf/commit/ed6788e710fc3093a7ecc2d078bf734c0f200d8d.

Reply:
The regression is that before the commit I linked, all the top-level metadata was added to the `localMeta` structure. After that commit, the delegations are not added anymore. The validation of roots, top targets and co created a sort of “whitelist” of which file can enter `localMeta` , and the delegated targets were not in that list.

The rest of the code basically assumes that if a metadata file is not in the `localMeta` , then it is not in the local storage and must be fetched from the remote storage. Which is problematic, because versions can be mismatched between local & remote, and the validation then fails due to the (local) snapshot not validating the (thus remote) delegated targets.

--- 

After discussing it with @trishankatdatadog, an idea of what to do to ensure validation would probably be to extract the `DelegatedTargets` code [from here](https://github.com/theupdateframework/go-tuf/blob/master/client/delegations.go#L12) to get a delegations verification DB from a given store, be it `LocalStore`, `RemoteStore`, or `localMeta`. We could then use it to verify what's added.

That's some work if we want to do it cleanly, and I won't have time to do it before EOY, so I align with @arbll here.